kernel/include/paging.h

```diff
@@ -34,23 +38,26 @@ extern uint64_t* page_zone_pointer;
 #define PAGETABLE_DEFAULT (PAGING_PRESENT | PAGING_WRITABLE)
 
 //returns the P4 pointer
-uint64_t* get_cr3();
+PageTable get_cr3();
 
 //sets the P4 pointer
-void set_cr3(uint64_t* P4);
+void set_cr3(PageTable P4);
+
 
+PageFlags get_page_flags(PageEntry page_entry);
+uint64_t get_page_address(PageEntry page_entry);
 
 //maps all virtual adidresses between virtaddr_low and virtaddr_high relatve to phys_low and makes sure they're in the pagetable
 //
 //ex:
 //map_pages(P4, 0x4000, 0x8000, 0x0000, PAGETABLE_DEFAULT)
 //will map all virtual addresses 0x4000 - 0x8000 to 0x0000 - 0x4000 using the flags in PAGETABLE_DEFAULT
-void map_pages(uint64_t* table, uint64_t virtaddr_low, uint64_t virtaddr_high, uint64_t phys_low, uint64_t flags);
+void map_pages(PageTable table, uint64_t virtaddr_low, uint64_t virtaddr_high, uint64_t phys_low, uint64_t flags);
 
 //identity maps any address between virtaddr_low and virtaddr_high and makes sure they're in the pagetable
 //all pages will have PAGETABLE_DEFAULT set 
-void identity_map_pages(uint64_t* table, uint64_t virtaddr_low, uint64_t virtaddr_high, uint64_t flags);
+void identity_map_pages(PageTable table, uint64_t virtaddr_low, uint64_t virtaddr_high, uint64_t flags);
 
 //gets a pointer to the pagetable entry that maps virtaddr to physical space
 //returns 0 if virtaddr is unpaged
-uint64_t* fetch_page(uint64_t* table, uint64_t virtaddr);
+PageEntry* fetch_page(PageTable table, uint64_t virtaddr);
```

Comment:
Maybe this function should be renamed because it fetches an entry and not an actual page. Maybe fetch_page_entry?

Reply:
I believe that we should use the contextual value of having PageEntry* as a return, instead of making excessively long names that 99% of the time will have the return type in its name.